internal/generate/tagstests/main.go

```diff
@@ -0,0 +1,235 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build generate
+// +build generate
+
+package main
+
+import (
+	_ "embed"
+	"fmt"
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/YakDriver/regexache"
+	multierror "github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-provider-aws/internal/generate/common"
+	"github.com/hashicorp/terraform-provider-aws/names/data"
+)
+
+func main() {
+	g := common.NewGenerator()
+
+	serviceData, err := data.ReadAllServiceData()
+
+	if err != nil {
+		g.Fatalf("error reading service data: %s", err)
+	}
+
+	servicePackage := os.Getenv("GOPACKAGE")
+
+	g.Infof("Generating tagging tests for internal/service/%s", servicePackage)
+
+	var (
+		serviceRecord data.ServiceRecord
+		found         bool
+	)
+
+	for _, l := range serviceData {
+		// See internal/generate/namesconsts/main.go.
+		p := l.ProviderPackage()
+
+		if p != servicePackage {
+			continue
+		}
+
+		serviceRecord = l
+		found = true
+		break
+	}
+
+	if !found {
+		g.Fatalf("service package not found: %s", servicePackage)
+	}
+
+	// Look for Terraform Plugin Framework and SDK resource and data source annotations.
+	// These annotations are implemented as comments on factory functions.
+	v := &visitor{
+		g: g,
+	}
+
+	v.processDir(".")
+
+	if err := v.err.ErrorOrNil(); err != nil {
+		g.Fatalf("%s", err.Error())
+	}
+
+	for _, foo := range v.taggedResources {
+		sourceName := foo.FileName
+		ext := filepath.Ext(sourceName)
+		sourceName = strings.TrimSuffix(sourceName, ext)
+		// filename := filepath.Join("./internal/service", servicePackage, fmt.Sprintf("%s_tags_test.go", sourceName))
+		filename := fmt.Sprintf("%s_tags_test.go", sourceName)
```

Comment:
Should the file name end with `_tags_gen_test.go` so that we can easily spot generated files?

Reply:
I went back and forth on the name. It's typically our convention, but combining `gen` and `test` seemed a bit ugly, especially since `test` has to go at the end. I'll change it